### ISSUE TITLE
[Feature Request] Allow complex search queries

### ISSUE THREAD

**Issue by raimund-schluessler** [nextcloud/tasks#471]
At the moment when a user types a string into the search input field, we search for this string in the summary, notes and categories.

I plan to implement to filter by certain properties, e.g. search for tags only. The query could look like "tags: Work, Accounting". This would also allow more complex queries filtering by more than one property like "tags: Work, priority: >5, due: >2019-12-24" or similiar.

**Comment by javisomoza** [comment]
+1

**Comment by davidszpak** [comment]
Plus, these queries would need to be savable as custom 'saved searches'.

**Comment by raimund-schluessler** [comment]
> Plus, these queries would need to be savable as custom 'saved searches'.

The corresponding issue for this idea is #220.

**Comment by samir-s** [comment]
Any update on this ? 

It would definitely allow any GTD users to use Nextcloud Task.

It would be awesome because believe it or not, there is no reliable webapp/desktop opensource GTD project syncing with a caldav server, with smart lists feature, AFAIK. 

The opensource Tasks.org on Android is the only one I know so far, it coud be a perfect match.

**Comment by meichthys** [comment]
I didn't realize it until now, but we can use the global search while looking at a task list to filter by tags. Just search for the tag name and it filters perfectly - no need to prepend anything like `tag: <tag_name>`

**Comment by Bugsbane** [comment]
This is true, however not an ideal solution. For example, I use tasks.org and have one label called "Reward" and another called "Rewardable". If I search for reward, it also brings up any task tagged as rewardable (as rewardable contains the string "reward").

It doesn't stop there though as if you're searching for reward, it isn't just going to bring up tasks labeled with that, but also any tasks that have the word reward elsewhere (eg in the description).
